src/Cohesity.Powershell/Scripts/Set-CohesityAlertResolutions.ps1

```diff
@@ -1,14 +1,45 @@
+function UpdateAlertResolution($alertIds, $resolvedAlertId) {
+    $url = $server + '/irisservices/api/v1/public/alertResolutions/'+$resolvedAlertId
+
+    $headers = @{'Authorization'='Bearer '+$token}
+    $payload = @{
+        alertIdList=@($alertIds)
+    }
+    $payloadJson = $payload  | ConvertTo-Json
+    $resp = Invoke-RestMethod -Method 'Put' -Uri $url -Headers $headers -Body $payloadJson -SkipCertificateCheck
+    $resolutionId = $resp.resolutionDetails.resolutionId
+    Write-Host "Successfully updated the resolution for the alert with id ="$resolutionId
+
+}
+function CreateAlertResolution($alertIds, $resolutionSummary, $resolutionDetails) {
+    $url = $server + '/irisservices/api/v1/public/alertResolutions'
+
+    $headers = @{'Authorization'='Bearer '+$token}
+    $payload = @{
+        alertIdList=@($alertIds)
+        resolutionDetails=@{
+            resolutionDetails=$resolutionDetails
+            resolutionSummary=$resolutionSummary
+        }
+    }
+    $payloadJson = $payload  | ConvertTo-Json
+    # Write-Host $payloadJson
+    $resp = Invoke-RestMethod -Method 'Post' -Uri $url -Headers $headers -Body $payloadJson -SkipCertificateCheck
+    $resolutionId = $resp.resolutionDetails.resolutionId
+    Write-Host "Successfully created a resolution for the alert with id ="$resolutionId
+
+}
 function Set-CohesityAlertResolutions {
     [CmdletBinding()]
     Param(
-        [Parameter(Mandatory = $true)]
-        [String]$resolvedAlertId,
-        [Parameter(Mandatory = $true)]
-        [String[]]$alertIds,
         [Parameter(Mandatory = $false)]
-        [String]$server,
+        [String]$resolvedAlertId = "",
```

Comment:
why should this be "" ? 
and whats the differens between resolvedAlertId and alertIds, wouldnt that be confusing to the user ?

Reply:
Fixed. Need to write modular components. Removing the additional functions and writing the inline codes. This is a temporary solution, need more reusable and sharable code in future.

Reply:
Fixed. Avoid using strong data type to leverage the usage of $null

Reply:
Renamed the parameters as ResolutionId, AlertIds, ResolutionSummary, ResolutionDetails